os/arch/arm/src/armv7-m/up_memfault.c

```diff
@@ -109,35 +113,36 @@ int up_memfault(int irq, FAR void *context, FAR void *arg)
 	/* Dump some memory management fault info */
 
 	(void)irqsave();
-	lldbg("PANIC!!! Memory Management Fault:\n");
-	mfdbg("  IRQ: %d context: %p\n", irq, regs);
-	lldbg("  CFAULTS: %08x MMFAR: %08x\n",
-		  getreg32(NVIC_CFAULTS), getreg32(NVIC_MEMMANAGE_ADDR));
-	mfdbg("  BASEPRI: %08x PRIMASK: %08x IPSR: %08x CONTROL: %08x\n",
-		  getbasepri(), getprimask(), getipsr(), getcontrol());
-	mfdbg("  R0: %08x %08x %08x %08x %08x %08x %08x %08x\n",
-		  regs[REG_R0], regs[REG_R1], regs[REG_R2], regs[REG_R3],
-		  regs[REG_R4], regs[REG_R5], regs[REG_R6], regs[REG_R7]);
-	mfdbg("  R8: %08x %08x %08x %08x %08x %08x %08x %08x\n",
-		  regs[REG_R8], regs[REG_R9], regs[REG_R10], regs[REG_R11],
-		  regs[REG_R12], regs[REG_R13], regs[REG_R14], regs[REG_R15]);
-
-#ifdef CONFIG_ARMV7M_USEBASEPRI
-#ifdef REG_EXC_RETURN
-	mfdbg("  xPSR: %08x BASEPRI: %08x EXC_RETURN: %08x (saved)\n",
-		  current_regs[REG_XPSR], current_regs[REG_BASEPRI], current_regs[REG_EXC_RETURN]);
-#else
-	mfdbg("  xPSR: %08x BASEPRI: %08x (saved)\n",
-		  current_regs[REG_XPSR], current_regs[REG_BASEPRI]);
-#endif
-#else
-#ifdef REG_EXC_RETURN
-	mfdbg("  xPSR: %08x PRIMASK: %08x EXC_RETURN: %08x (saved)\n",
-		  current_regs[REG_XPSR], current_regs[REG_PRIMASK], current_regs[REG_EXC_RETURN]);
-#else
-	mfdbg("  xPSR: %08x PRIMASK: %08x (saved)\n",
-		  current_regs[REG_XPSR], current_regs[REG_PRIMASK]);
-#endif
+	uint32_t *regs = (uint32_t *)context;
+	uint32_t cfsr = getreg32(NVIC_CFAULTS);
+	uint32_t mmfar = getreg32(NVIC_MEMMANAGE_ADDR);
+	lldbg("PANIC!!! Memory Management Fault occured while executing instruction at address : 0x%08x\n", regs[REG_R15]);
+	lldbg("CFAULTS: 0x%08x\n", cfsr);
+
+	if (cfsr & MMARVALID) {
+		lldbg("Access violation occured at address (MMFAR) : 0x%08x\n", mmfar);
```

Comment:
You can find belows at start of the up_assert.c file.
```
#ifdef CONFIG_ARCH_STACKDUMP
#undef  CONFIG_DEBUG
#undef  CONFIG_DEBUG_ERROR
#undef  CONFIG_DEBUG_WARN
#undef  CONFIG_DEBUG_VERBOSE
#undef  CONFIG_LOGM
#define CONFIG_DEBUG 1
#define CONFIG_DEBUG_ERROR 1
#define CONFIG_DEBUG_WARN 1
#define CONFIG_DEBUG_VERBOSE 1
#endif
```
Because of this, regardless of debug config (whether config_Debug is enabled or not), we can see crash messages.
But here is nothing like above. Then, this will not print the messages when config_debug and few configs are disabled.
Is this intended?

Reply:
@sunghan-chang 
I feel its better to get these logs always. So we will add below lines in these three fault handler files also.

Reply:
Done.